clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -543,6 +544,41 @@ object GarbageCollector {
     (gcAddressesLocation, gcCommitsLocation, expiredAddresses, runID)
   }
 
+  private def getPreviousRunID(
+      incrementalRun: Boolean,
+      storageNS: String,
+      configMapper: ConfigMapper
+  ): String = {
+    if (!incrementalRun) {
+      return ""
+    }
+    var previousRunID = ""
+    val runIDsPath = new Path(s"${storageNS.stripSuffix("/")}/_lakefs/retention/gc/run_ids/")
+    val runIDsFS = runIDsPath.getFileSystem(configMapper.configuration)
+    val runIDs = runIDsFS.listFiles(runIDsPath, false)
+    if (runIDs.hasNext) {
+      previousRunID = runIDs.next().getPath.getName
+      println(s"----------------------- Using previous RUN ID: $previousRunID")
+    } else {
+      runIDsFS.close()
+      throw RunIDException("No previous run ID")
```

Comment:
As I suggest in #5867, here too I prefer simply to run non-incrementally.  Think of the first GC run on a repo: no reason to fail it.  (Especially true in lakeFS.Cloud or other managed GC situations).

Reply:
lets discuss that on #5867